src/plugins/WebOfTrust/util/TickerDelayedBackgroundJob.java

```diff
@@ -0,0 +1,363 @@
+     * {@link InterruptedException}.
+     * @param job the job to run in the background
+     * @param name a human-readable name for the job
+     * @param delay the default background job aggregation delay in milliseconds
+     * @param ticker an asynchronous ticker with asynchronous executor
+     *
+     * @see TickerDelayedBackgroundJobFactory
+     */
+    public TickerDelayedBackgroundJob(Runnable job, String name, long delay, Ticker ticker) {
+        if (job == null || name == null || ticker == null || ticker.getExecutor() == null) {
+            throw new NullPointerException();
+        }
+        if (delay < 0) {
+            delay = 0;
+        }
+        this.realJob = new DelayedBackgroundRunnable(job);
+        this.name = name;
+        this.defaultDelay = delay;
+        this.ticker = ticker;
+        this.executor = ticker.getExecutor();
+    }
+
+    /**
+     * Triggers scheduling of the job with the default delay if no job is scheduled. If a job
+     * is already scheduled later than the default delay, it is rescheduled at the default delay.
+     *
+     * The first trigger received after the start of the last job execution leads to scheduling of
+     * another execution of the job, either after the default delay or when the currently executing
+     * job is finished, whichever comes last. A newly constructed delayed background job can be
+     * assumed to have started its last job infinitely in the past.
+     * @see #triggerExecution(long)
+     */
+    @Override
+    public synchronized void triggerExecution() {
+        tryEnqueue(defaultDelay);
+    }
+
+    /**
+     * Triggers scheduling of the job with the given delay if no job is scheduled. If a job
+     * is already scheduled later than the given delay, it is rescheduled at the given delay.
+     * Negative delays are treated as to zero delays.
+     *
+     * The first trigger received after the start of the last job execution leads to scheduling of
+     * another execution of the job, either after the default delay or when the currently executing
+     * job is finished, whichever comes last. A newly constructed delayed background job can be
+     * assumed to have started its last job infinitely in the past.
+     * @param delay the maximum trigger aggregation delay in milliseconds
+     * @see #triggerExecution()
+     */
+    @Override
+    public synchronized void triggerExecution(long delay) {
+        tryEnqueue(delay);
+    }
+
+    @Override
+    public synchronized void terminate() {
+        switch (state) {
+            case TERMINATED:
+                assert(waitingTickerJob == null) : "having ticker job in TERMINATED state";
+                assert(thread == null) : "having job thread in TERMINATED state";
+                return;
+            case TERMINATING:
+                assert(waitingTickerJob == null) : "having ticker job in TERMINATING state";
+                assert(thread != null) : "TERMINATING state but no thread";
```

Comment:
Also FYI, not a blocker: The existing strings merely put the variable status in human words. I usually only add strings if they add additional information which help the understanding. Asserts are typically only enabled by developers while having the source code at hand, so they don't need a message just for the sake of having a message.